core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -134,6 +138,7 @@ private[history] class FsHistoryProvider(conf: SparkConf) extends ApplicationHis
         TimeUnit.MILLISECONDS)
 
       if (conf.getBoolean("spark.history.fs.cleaner.enabled", false)) {
+        appsToClean = new ListBuffer[FsApplicationHistoryInfo]
```

Comment:
I'd just make `appsToClean` a `val` and initialize it regardless of this option. It's just a few bytes wasted when the cleaner is not enabled.